### PR TITLE
[Feature #5803] Add UnneededCondition cop

### PR DESCRIPTION
**Pull request by balbesina**
New cop `Style/UnneededCondition`. As described in #5803 it checks for same _condition_ and _if_branch_
```ruby
# bad
a = b ? b : c
      ^^^^^ Use double pipes `||` instead.
```
and suggests to replace ternary if `?:` with double pipes `||`.
```ruby
# good
a = b || c
```
Cop also checks regular `IfNode`s
```ruby
# bad
if b
  b
else
  c if d
end

# good
b || (c if d)
```

-----------------

Before submitting the PR make sure the following are checked:

* [x] Wrote [good commit messages][1].
* [x] Commit message starts with `[Fix #issue-number]` (if the related issue exists).
* [x] Feature branch is up-to-date with `master` (if not - rebase it).
* [x] Squashed related commits together.
* [x] Added tests.
* [x] Added an entry to the [Changelog](../blob/master/CHANGELOG.md) if the new code introduces user-observable changes. See [changelog entry format](../blob/master/CONTRIBUTING.md#changelog-entry-format).
* [x] The PR relates to *only* one subject with a clear title
  and description in grammatically correct, complete sentences.
* [x] Run `rake default` or `rake parallel`. It executes all tests and RuboCop for itself, and generates the documentation.

[1]: http://chris.beams.io/posts/git-commit/

### CODE REVIEW

**Comment by bbatsov** [comment]
It might not seem obvious, but such a cop should also handle `if/else`, as it's pretty much the same thing as a ternary op.

File: spec/rubocop/cop/style/unneeded_ternary_spec.rb [hunk 1]
```diff
@@ -0,0 +1,73 @@
+# frozen_string_literal: true
+
+RSpec.describe RuboCop::Cop::Style::UnneededTernary do
+  subject(:cop) { described_class.new }
+
+  it 'registers no offense for vars' do
+    expect_no_offenses('a = b ? d : c')
```
Comment (bbatsov) [inline spec/rubocop/cop/style/unneeded_ternary_spec.rb]: Not sure why all the example are using some assignment - obviously the assignment doesn't play any part in what the cop is checking for.

**Comment by balbesina** [comment]
I thought about regular `if` either. The problem is that _else_block_ might be complex (multiline) and I dont like the idea to change
```
if a
  a
else
  b
  c
end
```
into
```
a || begin
       b
       c
     end
```
What if I modify the cop to warn on unnecessary `if-else` when else block  is one-liner?

**Comment by bbatsov** [comment]
Makes sense to me.

**Comment by balbesina** [comment]
Cop's was modified. Now it also warns and corrects regular IfNode, when _else_branch_ is not too complex. e.g.
```ruby
# bad
if a
  b
else
  c if d
end

# good
a || (c if d)
```

**Pull request merged**